File: javafiles/day7_javafiles/instanceofex/Emp.java
package instanceofex;

public class Emp {
    private int empId;
    private String empName;
    private double sal;

    public Emp(int empId,String empName,double sal) {
        this.empId = empId;
        this.empName = empName;
        this.sal = sal;
    }

    public int getEmpId() {
        return empId;
    }

    public void setEmpId(int empId) {
        this.empId = empId;
    }

    public double computeAnnSal()
    {
        return this.sal * 12;
    }
}

class Manager extends Emp
{
    int teamSize;
    double bonus;

    public Manager(int empId, String empName,double sal,int teamSize, double bonus) {
        super(empId,empName,sal);
        this.teamSize = teamSize;
        this.bonus = bonus;
    }

    @Override
    public double computeAnnSal()
    {
        return super.computeAnnSal() + this.bonus;
    }

    public void delegate()
    {
        System.out.println("delegating to team of size:"+this.teamSize);
    }
}


class WageEmp extends Emp
{
    int noOfHrs;
    double dailyWages;

    public WageEmp(int empId, String empName, double sal, int noOfHrs, double dailyWages) {
        super(empId, empName, sal);
        this.noOfHrs = noOfHrs;
        this.dailyWages = dailyWages;
    }
}

class Main
{
    public static void main(String[] args) {
        Manager m = new Manager(2,"Shyam",10000,6,50000);
        m.computeAnnSal();
        m.delegate();



        Emp e = new Manager(1,"Ram",5000,3,10000);

        e.computeAnnSal();
        ((Manager)e).delegate();

        meth(e);
        meth(m);
        meth(new WageEmp(11,"Radha",5000,50,100));

    }


    static public void meth(Emp e)
    {
        e.computeAnnSal();

        if(e instanceof  Manager)
        {
            ((Manager)e).delegate();
        }
    }
}

File: react/my_app1/src/Emp.js

function Emp(props)
{
   
    const e = {empid:1,empname:'Ram',empsal:5000};
    const e1 = {empid:props.empid ,empname:props.empname ,empsal:props.empsal};


    return(
 
        <>
            <h3>empid = {e.empid}, empname = {e.empname},empsal = {e.empsal}</h3>
            <h3>empid = {e1.empid}, empname = {e1.empname},empsal = {e1.empsal}</h3>
            <ul>
                <li>empid={e1.empid}</li>
                <li>empname={e1.empname}</li>
                <li>empsal={e1.empsal}</li>
            </ul>
        
        </>
   
    )
}
export default Emp;